WcaOnRails/db/migrate/20231103103433_add_user_groups.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class AddUserGroups < ActiveRecord::Migration[7.0]
+  def change
+    delegate_regions_user_group = UserGroup.create!(name: "Delegate Regions", group_type: :delegate_regions, is_active: true, is_hidden: false)
+    senior_delegates = User.where(delegate_status: "senior_delegate")
+    senior_delegates.each do |delegate|
+      region = delegate.location
+      if region.include?("(")
+        region = region[0, region.index("(")]
+      end
+      region = region.strip
+      user_group = UserGroup.create!(name: region, group_type: :delegate_regions, parent_group_id: delegate_regions_user_group.id, is_active: true, is_hidden: false)
```

Comment:
What is the reasoning behind having a shared parent group for all Delegates?
Currently in the real-life WCA (not code), Senior Delegate is the highest rank. There is no "World Delegate" or similar, so I'm wondering why you chose to create a global Delegate user group.

Reply:
I guess as per our discussion, the user_groups will have the following on highest level: Delegate probation, Delegate regions, Teams, Committees, Councils, Banned group, Translators, etc. We can have all Delegate regions at top level, but the topmost level might have lots of data. Will it be fine if we have a highest level of Delegate regions so that the highest level looks clean?

Reply:
I'm not sure whether we're on the same page about the meaning of "groups on the highest level".

Your list is accurate and it reflects what we agreed on. So that's why we have the `group_type` column, right? I mean, you can just look at that column to figure out which groups are Delegate regions, and if we implement the Senior regions as the "highest" rank (which matches real-life WCA currently) then setting the senior region's parent_group to nil should be fine.

What do you mean by...?
> but the topmost level might have lots of data

Reply:
Ha okay, sorry for making it confused. As you said, group_type was the one that decides where it belongs. So having the highest level regions with parent_id null is better. I've changed it now.